### PR TITLE
document #17208

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

File: lib/pure/parseutils.nim [hunk 1]
```diff
@@ -164,8 +164,10 @@ proc parseHex*[T: SomeInteger](s: string, number: var T, start = 0,
   ## or to the end of the string. Otherwise, no more than ``maxLen`` characters
   ## are parsed starting from the ``start`` position.
   ##
-  ## It does not check for overflow. If the value represented by the string is
-  ## too big to fit into ``number``, only the value of last fitting characters
+  ## .. warning:: It does not check for overflow.
+  ## 
+  ## If the value represented by the string is too big to fit into ``number``, 
+  ## only the value of last fitting characters
```
Comment (konsumlamm) [inline lib/pure/parseutils.nim:170]:
```suggestion
  ## only the value of the last fitting characters
```

File: lib/pure/strutils.nim [hunk 2]
```diff
@@ -1148,6 +1148,16 @@ func parseHexInt*(s: string): int {.rtl, extern: "nsuParseHexInt".} =
   ## If `s` is not a valid hex integer, `ValueError` is raised. `s` can have one
   ## of the following optional prefixes: `0x`, `0X`, `#`.  Underscores
   ## within `s` are ignored.
+  ## 
+  ## .. warning:: It does not check for overflow.
+  ##
+  ## If the value represented by the string is too big to fit into `int`,
+  ## only the value of last fitting characters
```
Comment (konsumlamm) [inline lib/pure/strutils.nim:1155]:
```suggestion
  ## only the value of the last fitting characters
```

**Review by konsumlamm — CHANGES_REQUESTED** [review]

**Comment by timotheecour** [comment]
This is just too weird, we should fix the bug (https://github.com/nim-lang/Nim/issues/4350 and https://github.com/nim-lang/Nim/issues/17208) instead of documenting a buggy non-intuitive problem-hiding behavior for no benefit.
That comment https://github.com/nim-lang/Nim/issues/4350#issuecomment-226495965 is from 2016 and today's nim is rightfully more strict about correctness.

**Comment by juancarlospaco** [comment]
Fixing the Bug would be better, doc looks Ok.

**Comment by ringabout** [comment]
it needs to be fixed instead

**Pull request closed without merging**